tests/test_datasets/test_hierarchical_structure.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict
+from typing import List
+
+import numpy as np
+import pytest
+
+from etna.datasets import HierarchicalStructure
+
+
+@pytest.fixture
+def simple_hierarchical_struct():
+    return HierarchicalStructure({"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d"]}, ["l1", "l2", "l3"])
+
+
+@pytest.fixture
+def tailed_hierarchical_struct():
+    return HierarchicalStructure(
+        {"total": ["X", "Y"], "X": ["a"], "Y": ["c", "d"], "c": ["f"], "d": ["g"], "a": ["e", "h"]},
+        ["l1", "l2", "l3", "l4"],
+    )
+
+
+@pytest.mark.parametrize(
+    "target,source,answer",
+    (
+        ("l1", "l2", np.array([[1, 1]])),
+        ("l1", "l4", np.array([[1, 1, 1, 1]])),
+        ("l2", "l3", np.array([[1, 0, 0], [0, 1, 1]])),
+    ),
+)
+def test_tailed_struct_matrix(
```

Comment:
Also add separate test that checks the output format for the correctness(shape + type)